sys/drv/evdev.c

```diff
@@ -440,26 +454,18 @@ static int evdev_ioctl(file_t *f, u_long cmd, void *data) {
   return EINVAL;
 }
 
-static fileops_t evdev_fileops = {
-  .fo_read = evdev_read,
-  .fo_write = nowrite,
-  .fo_close = default_vnclose,
-  .fo_seek = noseek,
-  .fo_stat = default_vnstat,
-  .fo_ioctl = evdev_ioctl,
-};
-
-static int evdev_open(vnode_t *v, int mode, file_t *fp) {
-  evdev_dev_t *evdev = devfs_node_data(v);
-  int error = 0;
-
-  if ((error = vnode_open_generic(v, mode, fp)))
-    return error;
+static int evdev_open(devnode_t *master_dev, file_t *fp, int oflags) {
```

Comment:
You should reject attempts to open the device node when user passes `O_WRONLY` or `O_RDWR` to `open(2)`.

Reply:
Done!